docs(other-api/dev-v2): update MSW explanation (#6830)
remix-run/remix: docs/other-api/dev-v2.md

1. Run MSW as part of your app server
2. Configure MSW to not mock internal "dev ready" messages to the dev server

> For example, if you are using [binode][binode] to integrate with MSW,

Reviewer: Could you also remove the `[binode]` link definition at the bottom of this file since it would no longer be used?

{
  "scripts": {

>     "dev": "NODE_OPTIONS='--require ./mocks' remix dev",

Reviewer: I like the `NODE_OPTIONS` approach better for beginners, so overall like this change. But couple issues I see here:

1. Need to add `cross-env` for Windows support
2. This applies the `NODE_OPTIONS` to both `remix dev` (dev server) and the underlying app server (in this case, `remix-serve`, since the `-c` flag was omitted) which causes both processes to require the mocks even though only the app server needs it.

```suggestion
    "dev": "remix dev -c \"npm run dev:app\"",
    "dev:app": "cross-env NODE_OPTIONS=\"--require ./mocks\" remix-serve ./build"
```

1. Run MSW as part of your app server
2. Configure MSW to not mock internal "dev ready" messages to the dev server

> doing the following will give MSW access to the `REMIX_DEV_HTTP_ORIGIN` environment variable:

Reviewer: Make sure that you are setting up your mocks for your _app server_ within the `-c` flag so that the `REMIX_DEV_HTTP_ORIGIN` environment variable is available to your mocks:

> doing the following will give MSW access to the `REMIX_DEV_HTTP_ORIGIN` environment variable:

Reviewer: ```suggestion
Make sure that you are setting up your mocks for your _app server_ within the `-c` flag so that the `REMIX_DEV_HTTP_ORIGIN` environment variable is available to your mocks:
```

1. Run MSW as part of your app server
2. Configure MSW to not mock internal "dev ready" messages to the dev server

doing the following will give MSW access to the `REMIX_DEV_HTTP_ORIGIN` environment variable:

```json filename=package.json
{
  "scripts": {
    "dev": "remix dev -c \"npm run dev:app\"",
    "dev:app": "cross-env NODE_OPTIONS=\"--require ./mocks\" remix-serve ./build"

>     }

Author: ```suggestion
  }
```
